### PR TITLE
#6521 Exception when updating unpaid order

### LINKED ISSUE #6521: [Order] Status code 500 when updating unpaid order
**Steps to reproduce**
1. Sign in as a user.
2. Create a new order but leave the payment service page without completing the payment.
3. Navigate to your orders page.
4. Click the "Pay" button for the new unpaid order.
5. When reaching the final step of updating your order and click the "Pay" button the endpoint "/ubs/processOrder/{id}" will return a response with a 500 status code.

**Actual result**
You receive a response with a 500 status code, and a link to the payment page is not generated.

**Expected result**
You should receive a response with a 200 status code. The updated order should be saved, and the user should be redirected to the payment service.

### PR DESCRIPTION
**Pull request by Maksym-Lenets**
## Summary Of Issue :
The endpoint "/ubs/processOrder/{id}" should return a link that redirects users to the payment service page. However, the request failed during the order update process due to an issue with the update of the 'orderedBags' field.

**Issue Link**
https://github.com/ita-social-projects/GreenCity/issues/6521

## Summary Of Changes :
1) Added a helper methods to the Order entity and created corresponding tests.
2) Updated the usage of the 'orderedBags' field.
3) Updated the 'processOrder' method of the OrderController and created tests."

## CHECK LIST
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells or duplication
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: core/src/test/java/greencity/controller/OrderControllerTest.java [hunk 1]
```diff
@@ -160,10 +167,44 @@ void processOrder() throws Exception {
...
+        OrderDetailStatusDto orderDetailStatusDto = getUnpaidOrderDetailStatusDto();
+        orderDetailStatusDto.setOrderStatus(OrderStatus.FORMED.name());
+
+        FondyOrderResponse resultObject = FondyOrderResponse.builder()
+            .orderId(orderId)
+            .link("Link")
+            .build();
+        String resultJson = objectMapper.writeValueAsString(resultObject);
+
+        when(userRemoteClient.findUuidByEmail((anyString()))).thenReturn(uuid);
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/OrderControllerTest.java]: please add `verify`

File: core/src/test/java/greencity/controller/OrderControllerTest.java [hunk 2]
```diff
@@ -178,6 +219,32 @@ void processPaidOrderId() throws Exception {
...
+    @EnumSource(value = OrderStatus.class,
+        names = "FORMED",
+        mode = EnumSource.Mode.EXCLUDE)
+    void processPaidOrderIdWithUnacceptableOrderStatusesTest(OrderStatus orderStatus) throws Exception {
+        Long orderId = 1L;
+        OrderResponseDto dto = ModelUtils.getOrderResponseDto();
+        OrderDetailStatusDto orderDetailStatusDto = ModelUtils.getPaidOrderDetailStatusDto();
+        orderDetailStatusDto.setOrderStatus(orderStatus.name());
+
+        when(userRemoteClient.findUuidByEmail((anyString()))).thenReturn("35467585763t4sfgchjfuyetf");
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/OrderControllerTest.java]: please add `verify`

File: core/src/main/java/greencity/controller/OrderController.java [hunk 3]
```diff
@@ -175,7 +176,8 @@ public ResponseEntity<FondyOrderResponse> processOrder(
         @Valid @PathVariable("id") Optional<Long> id) {
         if (id.isPresent()) {
             OrderDetailStatusDto orderDetailStatusDto = ubsManagementService.getOrderDetailStatus(id.get());
-            if (orderDetailStatusDto.getPaymentStatus().equals("PAID")) {
+            if (orderDetailStatusDto.getPaymentStatus().equals("PAID")
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/OrderController.java]: please use `"PAID".equals(orderDetailStatusDto.getPaymentStatus())` to ommit `NPE`
and please move `PAID` to the constant

File: core/src/main/java/greencity/controller/OrderController.java [hunk 4]
```diff
@@ -175,7 +176,8 @@ public ResponseEntity<FondyOrderResponse> processOrder(
...
+                || !orderDetailStatusDto.getOrderStatus().equals(OrderStatus.FORMED.name())) {
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/OrderController.java]: same here

**Review by RomaMocherniuk — APPROVED** [review]

**Review by HelenSotnik — APPROVED** [review]

**Review by 2or5 — APPROVED** [review]

**Review by Yurii-Midianyi — APPROVED** [review]

**Pull request merged**